zerver/tests/test_project.py

```diff
@@ -0,0 +1,30 @@
+import importlib
+from unittest import mock
+
+from zerver.lib.test_classes import ZulipTestCase
+from zproject import config
+
+
+class ConfigTest(ZulipTestCase):
+    def test_get_mandatory_secret_succeed(self) -> None:
+        secret = config.get_mandatory_secret("shared_secret")
+        self.assertGreater(len(secret), 0)
+
+    def test_get_mandatory_secret_failed(self) -> None:
+        with self.assertRaises(config.ZulipSettingsError) as err:
+            config.get_mandatory_secret("nonexistent")
+        self.assertIn("nonexistent", str(err.exception))
+
+    def test_disable_mandatory_secret_check(self) -> None:
+        # Because we mock the environment variable, which affects how zproject.config.DISABLE_MANDATORY_SECRET_CHECK
+        # is set, we need to reload the module to see the change.
+        try:
+            with mock.patch.dict(
+                "zproject.config.os.environ", {"DISABLE_MANDATORY_SECRET_CHECK": "True"}
```

Comment:
This is a confusing equivalent of `mock.patch.dict(os.environ, {…})`—it mutates the actual `os.environ` dict, leaving both the `environ` binding in `os` and the `os` binding in `zproject.config` untouched.

(When patching a binding, e.g. `mock.patch("zerver.decorator.rate_limit_user", …)`, we need to specify the module where it’s bound, because the module object is what’s being mutated. This is not one of those times.)